src/pykeen/models/predict.py

```diff
@@ -511,3 +512,83 @@ def _process_remove_known(df: pd.DataFrame, remove_known: bool, testing: Optiona
     df = df[~df['in_testing']]
     del df['in_testing']
     return df
+
+
+@torch.no_grad()
+def _predict_triples(
```

Comment:
does this relate to any of the other functions in the predict.py module?

Reply:
The other ones do not use `predict_hrt` :wink: 

I think we could come up with some similar `batch_size` tuning code for the others, but then again, we already have something there from model training / evaluation, which catches more edge cases than the simple code I added here, so the way to go would be to figure out how we can re-use this part.

This might get a bit involved since I remember that @lvermue had to investigate a lot into when you manually need to clear caches etc.

We could also remove the tuning part here and have another separate issue for generalizing the AMO part.

Reply:
> The other ones do not use predict_hrt 😉

Okay

> I think we could come up with some similar batch_size tuning code for the others, but then again, we already have something there from model training / evaluation, which catches more edge cases than the simple code I added here, so the way to go would be to figure out how we can re-use this part.
> 
> This might get a bit involved since I remember that @lvermue had to investigate a lot into when you manually need to clear caches etc.
>
> We could also remove the tuning part here and have another separate issue for generalizing the AMO part.

I'll leave that up to you. I think it's fine to keep here - this all makes sense and is written with nice docs :)